cmd/skopeo/utils.go

```diff
@@ -354,3 +412,19 @@ func adjustUsage(c *cobra.Command) {
 	c.SetUsageTemplate(usageTemplate)
 	c.DisableFlagsInUseLine = true
 }
+
+func checkExistAndMkdir(destination string) error {
+	_, err := os.Stat(destination)
+	if err == nil {
+		return errors.Errorf("Refusing to overwrite destination directory %q", destination)
+	}
+	if !os.IsNotExist(err) {
+		return errors.Wrap(err, "Destination directory could not be used")
+	}
+	// the directory holding the image must be created here
+	if err = os.MkdirAll(destination, 0755); err != nil {
+		return errors.Wrapf(err, "Error creating directory for image %s", destination)
```

Comment:
Given the “Refusing to overwrite” behavior and this text saying “for image %s”, this seems unlikely to be reused for anything else — it is rather specific to the `dir:` transport; so I’m not sure that splitting this out is worthwhile.

I guess it depends on if we can find a good clear name for this behavior. `checkExistAndMkdir` is not that — that feels like two completely separate operations collected together, and `checkExist` doesn’t indicate anything about the consequences. So, for now it seems to me that leaving this in-line is _much_ easier than finding a good name for a separate function, but there might well be a great name / way to express this that I’m missing.